pkg/reconciler/kubernetes/tektonpipeline/tektonpipeline.go

```diff
@@ -124,6 +128,26 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, tp *v1alpha1.TektonPipel
 	// Mark PreReconcile Complete
 	tp.Status.MarkPreReconcilerComplete()
 
+	if _, err := common.ConfigReady(r.configInformer); err != nil {
+		if err.Error() == common.ConfigNotReady {
+			labels := r.manifest.Filter(mf.ByKind("Namespace")).Resources()[0].GetLabels()
+
+			ns, err := r.kubeClientSet.CoreV1().Namespaces().Get(ctx, tp.GetSpec().GetTargetNamespace(), metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			for key, value := range labels {
+				ns.Labels[key] = value
+			}
+			_, err = r.kubeClientSet.CoreV1().Namespaces().Update(ctx, ns, metav1.UpdateOptions{})
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		r.manifest = r.manifest.Filter(mf.Not(mf.ByKind("Namespace")))
+	}
+
```

Comment:
1. move all this into a function (ref: https://github.com/tektoncd/operator/pull/563/files#r778478046)
2. this logic should run before the `r.extension.Prereconcile()`